### DISCUSSION TITLE
Any hope for easily unlocking bootloader on Xiaomi devices ?

### DISCUSSION THREAD

**Discussion by freechelmi** [openandroidinstaller-dev/openandroidinstaller#403] [General]
Hi , received my first xiaomi ( lancelot )

Unlocking Bootloader on Xiaomi devices is a nightmare : 

- Windows only official tool ( Unofficial XiaomiToolV2 seems dead ) 
- Need to insert a SIM and link MI account to use it 
- Seems to be limited to one device per SIM per Month 

Do we have any alternative to this ? 

Is there any current legal action against Xiaomi to change that  ?

**Comment by tsterbak** [comment]
Yeah, Xiaomi devices are a pain...

**Comment by ghost** [comment]
The benefit with MediaTek devices is [mtkclient](https://github.com/bkerler/mtkclient)
It worked great with my Redmi Note 10S (rosemary)

Installing custom ROM doesn't void the guarantee in the EU, but I don't know if there were discussions about unlocking phones...

**Reply by freechelmi** [reply]
Can you explain how mtkclient could allow us to unlock the bootloader or run untrusted rom ?

**Reply by ghost** [reply]
I don't think it's a good idea to use it on OAI (it uses SOC security flows, and can lead into bricking the device)
I just mentioned it because it works, especially on older (not patched) mtk devices

**Reply by freechelmi** [reply]
Lancelot redmi 9 is Mediatek MT6768 Helio G85  which seems supported by Mtkclient , I will give it a try , Xiaomi people are stupid , the client service on the phone even tell me that they don't allow unlocking the bootloader

**Comment by MagicLike** [comment]
There are some approaches in #286...

**Comment by freechelmi** [comment]
Also do we have some recovery for lancelot ( redmi 9 ) No TWRP or ofox it seems ?

**Reply by ghost** [reply]
This one, for example : https://xdaforums.com/t/recovery-twrp-unofficial-twrp-recovery-3-6-1_11-0.4421301/
And I think a lot more on lancelot telegram channels ( e.g. this one : https://t.me/redmi_9 )

**Reply by ghost** [reply]
Always look at Indonesian telegram channels, there is a lot of interesting stuff

**Reply by freechelmi** [reply]
So Far I could not find a working TWRP with android 12 on lancelot 

When I try to boot them I get FAILED (Status read failed (No such device))

**Comment by freechelmi** [comment]
Found this list that sum up what can be done for each device https://unlocktool.net/models/
